src/CachePlugin.php

```diff
@@ -36,11 +42,13 @@ class CachePlugin implements Plugin
 
     /**
      * @param CacheItemPoolInterface $pool
+     * @param StreamFactory          $streamFactory
      * @param array                  $options
      */
-    public function __construct(CacheItemPoolInterface $pool, array $options = [])
+    public function __construct(CacheItemPoolInterface $pool, StreamFactory $streamFactory, array $options = [])
```

Comment:
Ah, okey, sure. 

If we should not have the options array in the constructor, I would prefer to use multiple setter for each property: 

```php

    /**
     * @param boolean $respectCacheHeaders
     *
     * @return CachePlugin
     */
    public function setRespectCacheHeaders($respectCacheHeaders)
    {
        $this->respectCacheHeaders = $respectCacheHeaders;

        return $this;
    }

    /**
     * @param int $defaultTtl
     *
     * @return CachePlugin
     */
    public function setDefaultTtl($defaultTtl)
    {
        $this->defaultTtl = $defaultTtl;

        return $this;
    }
```